### PR TITLE
feat: added Home page components

### PR DESCRIPTION
**Pull request by sjwu1235**
# Pull Request

## Scope
Create a Homepage with a search bar
https://dev.azure.com/JHNMAT015/ECO5037S%20Front%20end/_workitems/edit/97 

## Work Done
Added two components Home.vue and SearchBar.vue (this can be reused for navbar in other pages) and accompanying tests. Added style sheet and updated App.vue to show placeholders for navbar and footer.
Limitations: Router link for logo is disabled, search button is not connected.

## Steps to Test
```npm run start```
Should see home page with search bar

## Screenshot
image

### CODE REVIEW

File: src/components/Home.vue [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  <div class="container">
+    <!--<router-link to="/">-->
+    <img id="homelogo" alt="Vue logo" src="../assets/logo_green.png" />
+    <!--</router-link>-->
+    <search-bar />
+  </div>
+</template>
+<script>
+//since using router need to make a search component
+console.log('Homepage loaded')
```
Comment (DominicGBauer) [inline src/components/Home.vue:11]: Delete this

File: src/components/Home.vue [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  components: {
+    SearchBar,
+  },
+}
+</script>
+<!-- Add "scoped" attribute to limit CSS to this component only -->
+<style scoped>
+#main {
+  min-height: 100%;
+  /*    display: grid;
```
Comment (DominicGBauer) [inline src/components/Home.vue:28]: Why is this whole comment here?

File: src/components/Home.vue [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  name: 'HomePage',
+  props: {
+    Search: String,
+  },
+  components: {
+    SearchBar,
+  },
+}
+</script>
+<!-- Add "scoped" attribute to limit CSS to this component only -->
```
Comment (DominicGBauer) [inline src/components/Home.vue:24]: You can delete this as this is known or at least it should be

File: src/components/SearchBar.vue [hunk 4]
```diff
@@ -0,0 +1,51 @@
+<template>
+  <div class="input-icons">
+    <input id="search" type="text" placeholder="Search" />
+    <i class="fa fa-search icon"></i>
+  </div>
+</template>
+<script>
+//since using router need to make a search component
+console.log('Searchbar loaded')
```
Comment (DominicGBauer) [inline src/components/SearchBar.vue:9]: delete this and comment above

File: src/components/SearchBar.vue [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+export default {
+  name: 'SearchBar',
+  props: {
+    Search: String,
+  },
+}
+</script>
+<!-- Add "scoped" attribute to limit CSS to this component only -->
```
Comment (DominicGBauer) [inline src/components/SearchBar.vue:18]: again delete this

File: src/components/SearchBar.vue [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+#search:focus {
+  box-shadow: 1px 1px 5px grey;
+  outline: none;
+}
+#search:hover {
+  box-shadow: 1px 1px 5px grey;
+}
+
+.input-icons {
+  /*display: inline-block;*/
```
Comment (DominicGBauer) [inline src/components/SearchBar.vue:37]: why is this here?

File: public/style.css [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    min-height: 100vh;
+    display: flex;
+    flex-direction: column;
+}
+footer{
+    position: absolute;
+    bottom: 0;
+}
+.container {
+    display: flex; /* or inline-flex */
```
Comment (DominicGBauer) [inline public/style.css:15]: delete the comment

File: src/App.vue [hunk 8]
```diff
@@ -1,15 +1,26 @@
 <template>
-  <img alt="Vue logo" src="./assets/logo_green.png" />
-  <HelloWorld msg="Welcome to Your Vue.js App" />
+  <div>
+    <nav id="navbar">
+      <h1>Placeholder for navbar</h1>
+      <!--possible approach: use same navbar for homepage as others and hide search bar when detected as home route-->
```
Comment (DominicGBauer) [inline src/App.vue:5]: delete this comment, code isn't the place to discuss what should go where

File: src/App.vue [hunk 9]
```diff
@@ -1,15 +1,26 @@
...
+    </nav>
+    <div id="main">
+      <!--<router-view />-->
```
Comment (DominicGBauer) [inline src/App.vue:8]: delete this

File: src/components/Home.vue [hunk 10]
```diff
@@ -0,0 +1,43 @@
+<template>
+  <div class="container">
+    <!--<router-link to="/">-->
```
Comment (DominicGBauer) [inline src/components/Home.vue:3]: delete this

File: src/components/Home.vue [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    <img id="homelogo" alt="Vue logo" src="../assets/logo_green.png" />
+    <!--</router-link>-->
```
Comment (DominicGBauer) [inline src/components/Home.vue:5]: delete this

File: src/components/Home.vue [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+    <search-bar />
+  </div>
+</template>
+<script>
+//since using router need to make a search component
```
Comment (DominicGBauer) [inline src/components/Home.vue:10]: delete this

**Review by DominicGBauer — CHANGES_REQUESTED** [review]

File: src/components/Home.vue (hunk 2 above)
Comment (sjwu1235) [reply src/components/Home.vue:28]: missed it when cleaning up.

File: src/components/SearchBar.vue [hunk 13]
```diff
@@ -0,0 +1,46 @@
+<template>
+  <div class="input-icons">
```
Comment (DominicGBauer) [inline src/components/SearchBar.vue:2]: there's only one icon right? Why is this pluralized? Also isn't this just the search-container?

File: src/components/SearchBar.vue [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+}
+#search:focus {
+  box-shadow: 1px 1px 5px grey;
+  outline: none;
+}
+#search:hover {
+  box-shadow: 1px 1px 5px grey;
+}
+
+.input-icons {
```
Comment (DominicGBauer) [inline src/components/SearchBar.vue:32]: You should use flexbox for this. Don't need position relative because it will confuse things later. Basically always use flexbox unless there's an incredibly good reason not to

File: src/components/SearchBar.vue [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+  width: 50%;
+  margin-bottom: 10px;
+  position: relative;
+  align-self: center;
+  height: 30px;
+}
+
+i {
+  position: absolute;
```
Comment (DominicGBauer) [inline src/components/SearchBar.vue:41]: Again use flexbox

**Review by DominicGBauer — CHANGES_REQUESTED** [review]

**Review by DominicGBauer — APPROVED** [review]
